genie-agent/src/main/java/com/netflix/genie/agent/execution/process/impl/JobProcessManagerImpl.java

```diff
@@ -197,8 +227,32 @@ public JobProcessResult waitFor() throws InterruptedException {
             // Do nothing.
         }
 
-        if (killed.get()) {
-            return new JobProcessResult.Builder(JobStatus.KILLED, JobStatusMessages.JOB_KILLED_BY_USER).build();
+        // If for whatever reason the timeout thread is currently running or if it is scheduled to be run, cancel it
+        final ScheduledFuture timeoutThreadFuture = this.timeoutKillThread.get();
+        if (timeoutThreadFuture != null) {
+            timeoutThreadFuture.cancel(true);
+        }
+
+        // TODO: This doesn't seem quite right to me and is confirmed by an existing test. If the job completes
+        //       successfully but then the system calls kill before this method then the final job status will be
+        //       killed instead of successful. We should discuss this - TJG 9/9/2019
```

Comment:
@mprimi I just wanted to call out this `TODO` to get your thoughts. We can discuss in person if easier

Reply:
I don't think it's a super big deal. The user called for a kill (or the timeout kicked in) just as the job was completing, then the final status is `killed`.
It's very unlikely to ever happen, but even if it does, it's not too incorrect, no? The job was killed before it had time to finish and finalize cleanly.